Assets/Scripts/Source/Game/Combat/CombatManager.cs

```diff
@@ -59,9 +83,122 @@ void generateEnemyActions()
     }
 
     IEnumerator executeActions()
-    {   
+    {
         while(!turnManager.ShouldWaitForPlayerAction(characterManager.GetActivePlayer()))
         {
+            float timer = 0.0f;
+            float missDelay = 0.0f;
+            bool missed = false;
+            CharacterAction action = turnManager.Peek();
+            CharacterAction enemyAction = action.Targets[0].ActiveAction; //TODO: Maybe add target enemy incase of multiple enemies
+            
+            if (action.Actor is Player)
+            {
+                if (action is Attack)
+                {
+                    //Debug.Log("Player turn attack");
+                    while (timer < attackTimer && !action.Actor.hitBonus)
```

Comment:
I wonder if we could break this listener logic off into a helper class, for example: `CharacterActionTimer` w/ params for the input/timeout/bonus on success ?